app/models/school_data_importer.rb

```diff
@@ -24,18 +24,48 @@ def schools_data_file
   end
 
   def row_to_school(row)
+    local_authority = LocalAuthority.find_or_initialize_by(code: row.fetch("LA (code)"))
+    row_local_authority_name = row.fetch("LA (name)")
+    if local_authority.persisted? && local_authority.name != row_local_authority_name
+      Rails.logger.info "LA name change in school import. Old name: #{local_authority.name}, New name: #{row_local_authority_name}"
+    end
+    local_authority.name = row_local_authority_name
+    local_authority.save!
+
+    local_authority_district = LocalAuthorityDistrict.find_or_initialize_by(code: row.fetch("DistrictAdministrative (code)"))
+    row_lad_name = row.fetch("DistrictAdministrative (name)")
+    if local_authority_district.persisted? && local_authority_district.name != row_lad_name
+      Rails.logger.info "LA name change in school import. Old name: #{local_authority_district.name}, New name: #{row_lad_name}"
+    end
+    local_authority_district.name = row_lad_name
+    local_authority_district.save!
+
```

Comment:
you could also spread the method into three sections - one handling local authority , one handling local authority district and one just managing the school data